pandas/core/indexes/interval.py

```diff
@@ -1003,6 +1004,47 @@ def get_loc(self, key, method=None):
             else:
                 return self._engine.get_loc(key)
 
+    def get_loc_exact(self, key, method=None):
+        """Get integer location, slice or boolean mask for exact
+        Interval matches only.
+
+        Parameters
+        ----------
+        key : Interval
+            The label we want to find locations for. Must have type
+            :class:`Interval`
+        method : {None}, optional
+            * default: matches where the label exactly matches a given
+              :class:`Interval`.
+
+        Returns
+        -------
+        loc : int if unique index, slice if monotonic index, else mask
+
+        Examples
+        ---------
+        >>> i1, i2 = pd.Interval(0, 1), pd.Interval(1, 2)
+        >>> index = pd.IntervalIndex([i1, i2])
+        >>> index.get_loc_exact(i1)
+        0
+
+        If an exact match is not found, a KeyError is raised
+        >>> index.get_loc_exact(pd.Interval(0.5, 1.5))
+        KeyError: Interval(0.5, 1.5, closed='right')
+
+        If a label is in several locations, you get all the relevant
+        locations.
+
+        >>> index = pd.IntervalIndex([i1, i2, i1])
+        >>> index.get_loc_exact(i1)
+        array([ True, False,  True], dtype=bool)
+
+        See Also
+        --------
+        get_loc
+        """
+        return self.astype(object).get_loc(key, method=method)
```

Comment:
this seems quite inefficient. a) can you add an asv, b) I would do this in the cython code itself that implements get_loc (this can be a further refinement / PR).

Reply:
The super method was faster, around 13 ms, while this is around 75 ms, so it's slower.

I don't have experience with cython, so can't take on those changes in get_loc, sorry.

Reply:
ok that's fine an asv for now is prob best.

Reply:
I've looked into asv and the speed issue. 

```python
>>> idx = IntervalIndex.from_breaks(np.arange(1000001))
>>> i = idx[80000]
>>> %t idx.get_loc(i)
25.2 µs ± 1.31 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
>>> %t idx.get_loc_exact(i)
1.33 s ± 46.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

I doubt this acceptable?

Reply:
wait, so this is *much* slower?

Reply:
Yes. I think this speed issue is beyound me, so I'm stuck. Maybe it's something wrong with Interval type conversion?

Reply:
so why don't you just do .get_loc and then check if its exact after?

Reply:
Ok, that works very nicely:  41.5 µs for ``get_loc_exact`` vs 33.1 µs for ``get_loc``.

so there's no speed issue now at all.